pytorch_lightning/core/lightning.py

```diff
@@ -298,19 +294,20 @@ def log(
 
         Args:
             name: key to log
-            value: value to log
+            value: value to log. Can be a ``float``, ``Tensor``, ``Metric``, or a dictionary of the former.
             prog_bar: if True logs to the progress bar
             logger: if True logs to the logger
             on_step: if True logs at this step. None auto-logs at the training_step but not validation/test_step
             on_epoch: if True logs epoch accumulated metrics. None auto-logs at the val/test step but not training_step
             reduce_fx: reduction function over step values for end of epoch. :meth:`torch.mean` by default.
             enable_graph: if True, will not auto detach the graph
             sync_dist: if True, reduces the metric across GPUs/TPUs
-            sync_dist_op: the op to sync across GPUs/TPUs
             sync_dist_group: the ddp group to sync across
             add_dataloader_idx: if True, appends the index of the current dataloader to
                 the name (when using multiple). If False, user needs to give unique names for
                 each dataloader to not mix values
+            batch_size: Current batch_size. This will be directly inferred from the loaded batch,
+                but some data structures might need to explicitly provide it.
```

Comment:
Here are the docs for batch size, is this what you mean?

Reply:
Yeah but what I'm missing is *why* e.g. what is that parameter used for (inferred or otherwise)?

Reply:
To compute the correct average when we ask `self.log` to average the metric on epoch end. 

It has to be weighted by the batch size because often the last batch does not have the same size as the others.
The dataset is not guaranteed to be divisible by the batch size and the `drop_last` in the PyTorch DataLoader is not `True` by default.

Reply:
In the ResultMetric in result.py you will find the line:

`self.cumulated_batch_size += batch_size
`
and the cumulated_batch_size is then used in the compute() method.

Reply:
ok that's what I thought it was for but I couldn't find in the code where it's actually doing that. 

So does this mean that my `_step` should log a scalar which is the mean of the current batch and PL will correctly average (including across DDP processes) by multiplying with the batch size, summing, then dividing by the dataset size?

Reply:
Yes, relevant code:

https://github.com/PyTorchLightning/pytorch-lightning/blob/cdcc483e9b7a79de3e5a7ac9c1e9dfd12ab77f4f/pytorch_lightning/trainer/connectors/logger_connector/result.py#L169-L197